memcache/provider.go

```diff
@@ -146,7 +155,9 @@ func (mcp *Provider) Regenerate(oldID, newID []byte) (session.Storer, error) {
 		newItem := acquireItem()
 		newItem.Key = newKey
 		newItem.Value = oldItem.Value
-		newItem.Expiration = oldItem.Expiration
+		// Expiration can be converted safely from time.Duration to in32 because
+		// mcp.expiration has been checked to be safely castable in Init method.
+		newItem.Expiration = int32(mcp.expiration / time.Second)
```

Comment:
it's necessary since the value stored is the default value which is 7200000000 and that does not fit in int32... Same issue for all providers, SQL providers cannot store such a value. It throws when running the example.

Reply:
So we either need to consider not using Duration anymore but an int32 representing the number of seconds or do those divisions. You choose.

Reply:
mcp.expiration previously check in Provider.Init o int set.Expiration, here it's not neccesary to divide by seconds, because the value already be correct

Reply:
the value is in microseconds and cannot be stored as an int in DB or cache backend, it's too big.

Reply:
look at what `2 * time.Hour` produces. This is the default value for the expiration and it's expressed in microsecond and that overflows an int32. It's not even comparable with a Unix timestamp as last_activity is stored in DB.

Reply:
`DELETE FROM %s WHERE last_active+expiration<=$1` does not work if not converted in seconds. The driver throws because the value is too big. Try it out with the example, you'll see.

Reply:
in SQL i agree, but in memcache you put a correct value on item, later the memcache library parse this value to the most convenient value for him

Reply:
Here for memcache the value should be expressed in seconds as well as per the documentation: https://github.com/golang/appengine/blob/master/memcache/memcache.go#L65

Reply:
The type of memcache.Item.Expiration is a time.Duration

Reply:
you're right, but this code is newer... In you're current version it's an int32...

Reply:
So my guess is there was a bug hidden by the fact that the cookie expires before the data. A bump of the version of memcache would fix that :).

Reply:
Oh of course, update the memcache library.

Reply:
well, the link I provided is not from the same library. In the one you're using it's actually an int32 even in master. https://github.com/bradfitz/gomemcache/blob/551aad21a6682b95329c1f5bd62ee5060d64f7e8/memcache/memcache.go#L167

Reply:
Yeah, i see after, but don't worry. let's leave it with the current library